desktop_version/src/Input.cpp

```diff
@@ -587,12 +587,55 @@ void menuactionpress()
         music.playef(11);
         if (loc::languagelist.size() != 0)
         {
-            loc::lang = loc::languagelist[game.currentmenuoption].code;
-            loc::loadtext();
-            game.savestats();
+            if ((unsigned)game.currentmenuoption < loc::languagelist.size())
+            {
+                loc::lang = loc::languagelist[game.currentmenuoption].code;
+                loc::loadtext();
+                game.savestats();
+                map.nexttowercolour();
+                game.returnmenu();
+            }
+            else if (loc::show_lang_maint_menu)
+            {
+                game.createmenu(Menu::language_maint);
+            }
+        }
+        else
+        {
+            map.nexttowercolour();
+            game.returnmenu();
+        }
+        break;
+    case Menu::language_maint:
+        music.playef(11);
+        switch (game.currentmenuoption)
+        {
+        case 0:
+            // sync languages
+            game.createmenu(Menu::language_maint_sync);
+            map.nexttowercolour();
+            break;
+        case 1:
+            // translation statistics
+            // TODO
+            map.nexttowercolour();
+            break;
+        case 2:
+            // return. Also leave the language selection menu
+            game.returnmenu(); game.returnmenu();
```

Comment:
You can do `game.returntomenu(Menu::options);` instead, if that fits your intent better.

Reply:
Sounds good, done in cb2986d7f3c1241be91bd0f81ee65e4b254270d0.